### DISCUSSION TITLE
The build configurations generated do not contain the active build configuration

### DISCUSSION THREAD

**Discussion by stayprivate** [microsoft/vscode-cmake-tools#2248] [Q&A] [answered]
Notice the message in Bold in the following output.

[variant] Loaded new set of variants
[kit] Successfully loaded 18 kits from 
[kit] Successfully loaded 5 kits from 
[cmakefileapi-driver] This version of CMake does not support the "toolchains" object kind. Compiler paths will be determined by reading CMakeCache.txt.
[main] Configuring folder: logc2 
[proc] Executing command: /usr/bin/cmake --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Release -DCMAKE_CC_COMPILER:FILEPATH=/usr/lib/icecc/bin/gcc-9 -DCMAKE_CXX_COMPILER:FILEPATH=/usr/lib/icecc/bin/g++-9 -H... B... -G Ninja
[cmake] Not searching for unused variables given on the command line.
[cmake] -- Using C++17
**[cpptools] The build configurations generated do not contain the active build configuration. Using 'Debug' for CMAKE_BUILD_TYPE instead of 'Release' to ensure that IntelliSense configurations can be found**
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: ,,,/bin/Release

cmake-variants.json does contain a defintion for Release and Debug. Using  v1.7.1 of cpptool and v1.9.1 of cmake tool. That's on Ubuntu 20

{
  "buildType": {
    "default": "Release",
    "description": "The CMake build type to use",
		"choices": {
			"Release:": {
				"short": "Release",
				"long": "Fast and Furious",
				"buildType": "Release"
			},
			"Debug": {
				"short": "Debug",
				"long": "Emit debug information",
				"buildType": "Debug"
			},
...

Suggestion?  

Thanks.
- Mario

**Comment by bobbrow** [comment, marked as answer]
This looks like https://github.com/microsoft/vscode-cmake-tools/issues/2120.  You can ignore the warning for now.  We will get it fixed in a future version of CMake Tools.

**Reply by bobbrow** [reply]
Actually, there's more going on here according to your log.  Let's transfer this discussion over to the CMake Tools project.

**Reply by bobbrow** [reply]
Can you run the `CMake: Log Diagnostics` command and share the output here?  It looks like you're trying to set the build type to "Release", but I suspect that something in your CMake script is changing it back to "Debug" if the extension is only getting compilation information for the Debug build type.
